src/misc.cpp

```diff
@@ -107,6 +111,22 @@ void InitializeGame(uint size_x, uint size_y, bool reset_date, bool reset_settin
 	_thd.redsq = INVALID_TILE;
 	if (reset_settings) MakeNewgameSettingsLive();
 
+	/* Set the Random() seed to generation_seed so we produce the same map with the same seed */
+	if (_settings_game.game_creation.generation_seed == GENERATE_NEW_SEED) _settings_game.game_creation.generation_seed = InteractiveRandom();
+
+	_random.SetSeed(_settings_game.game_creation.generation_seed);
+	ScriptObject::InitializeRandomizers();
+
+	/* Set random deviation for scripts. */
+	for (CompanyID c = COMPANY_FIRST; c < MAX_COMPANIES; c++) {
+		if (_settings_game.ai_config[c] != nullptr) _settings_game.ai_config[c]->AddRandomDeviation();
+	}
```

Comment:
Hmm maybe it would make more sense to apply random deviation when the AI starts (of course not when starting during loading)

Reply:
Not sure, you then don't see the correct value in the settings before it starts, and that might warrant a UI change.